pkg/processor/sourceprocessor/source_category_filler_test.go

```diff
@@ -131,4 +134,20 @@ func TestFillWithContainerAnnotations(t *testing.T) {
 
 		assertAttribute(t, attrs, "_sourceCategory", "THIRD_s-c!")
 	})
+
+	t.Run("set source category prefix", func(t *testing.T) {
```

Comment:
I'm not sure what is the intent of this test? It does not enable container annotations (`cfg.ContainerAnnotations.Enabled = true`), so it only tests that the source category configured in the processor's settings is used.

We should rather have a test that defines what happens if you try to use templates in per-container annotations. I propose something like this:

```go
	t.Run("container annotations do not support templating", func(t *testing.T) {
		cfg := createDefaultConfig().(*Config)
		cfg.ContainerAnnotations.Enabled = true
		cfg.SourceCategory = "my-source-category"
		cfg.SourceCategoryPrefix = "my-prefix"

		attrs := pcommon.NewMap()
		attrs.PutStr("k8s.container.name", "container-name-1")
		attrs.PutStr("k8s.namespace.name", "ns-1")
		attrs.PutStr("k8s.pod.name", "ABC")
		attrs.PutStr("k8s.pod.annotation.sumologic.com/sourceCategory", "sc-from-annot-%{k8s.namespace.name}-%{k8s.pod.uid}")
		attrs.PutStr("k8s.pod.annotation.sumologic.com/sourceCategoryPrefix", "%{k8s.pod.name}-%{k8s.pod.uid}-Prefix:")
		attrs.PutStr("k8s.pod.annotation.sumologic.com/sourceCategoryReplaceDash", "#")
		attrs.PutStr("k8s.pod.annotation.sumologic.com/container-name-1.sourceCategory", "%{k8s.namespace.name}-%{k8s.pod.name}")
		attrs.PutStr("k8s.pod.annotation.sumologic.com/container-name-2.sourceCategory", "another/source-category")
		attrs.PutStr("k8s.container.name", "container-name-1")

		filler := newSourceCategoryFiller(cfg, zap.NewNop())
		filler.fill(&attrs)

		assertAttribute(t, attrs, "_sourceCategory", "%{k8s.namespace.name}-%{k8s.pod.name}")
	})
```